### ISSUE TITLE
SpecifyCultureInfo (CA1304) reports on WinForms designer-generated code

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn-analyzers#951]
#### Analyzer package
System.Runtime.Analyzers

#### Analyzer
CA1304: SpecifyCultureInfo

#### Repro steps
1. Create a new WinForms application
2. Add a custom icon to the default Form
3. CA1304 is reported on designer-generated file Form1.Designer.cs:

```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CA1304	Because the behavior of 'System.Resources.ResourceManager.GetObject(string)' could vary based on the current user's locale settings, replace this call in 'WindowsFormsApplication1.Form1.InitializeComponent()' with a call to 'System.Resources.ResourceManager.GetObject(string, System.Globalization.CultureInfo)'. If the result of 'System.Resources.ResourceManager.GetObject(string, System.Globalization.CultureInfo)' will be displayed to the user, specify 'CultureInfo.CurrentCulture' as the 'CultureInfo' parameter. Otherwise, if the result will be stored and accessed by software, such as when it is persisted to disk or to a database, specify 'CultureInfo.InvariantCulture'.	WindowsFormsApplication1	e:\Bart\Documents\Visual Studio 2015\Projects\WindowsFormsApplication1\WindowsFormsApplication1\Form1.Designer.cs	39	Active
```

Generated code looks like:
```csharp
        private void InitializeComponent()
        {
            // ...

            this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));

            // ...
        }
```

Another case that generates CA1304 on a Form.Designer.cs file:
```csharp
this.soundPreviewButton.BackgroundImage = ((System.Drawing.Image)(resources.GetObject("soundPreviewButton.BackgroundImage")));
```
#### Expected behavior
No report, because the file is designer-generated.

#### Actual behavior
CA1304 is reported.

**Comment by mavasani** [comment]
Should get fixed with https://github.com/dotnet/roslyn-analyzers/pull/964

**Comment by bkoelman** [comment]
Confirmed this is now working correctly.
